public Object call() throws Exception {
    return action.run();
}